test-data/unit/check-type-aliases.test

```diff
@@ -796,3 +796,154 @@ S = TypeVar("S")
 class C(Generic[S], List[Defer]): ...
 class Defer: ...
 [builtins fixtures/list.pyi]
+
```

Comment:
These tests are maybe a little excessive; lmk if we want them pruned down a bit.

Reply:
They are maybe slightly excessive, but I think it's fine to keep them as is.